controllers/mustgather/mustgather_controller.go

```diff
@@ -367,23 +374,78 @@ func (r *MustGatherReconciler) addFinalizer(reqLogger logr.Logger, m *mustgather
 	return nil
 }
 
-func (r *MustGatherReconciler) getJobFromInstance(instance *mustgatherv1alpha1.MustGather) (*unstructured.Unstructured, error) {
+func (r *MustGatherReconciler) getJobFromInstance(instance *mustgatherv1alpha1.MustGather) (*batchv1.Job, error) {
 	version, err := r.getClusterVersionForJobTemplate("version")
 	if err != nil {
 		return nil, fmt.Errorf("failed to get cluster version for job template: %w", err)
 	}
 
-	jobTemplate, err = initializeTemplate(version)
+	jobTemplate, err := initializeTemplate(version)
 	if err != nil {
 		log.Error(err, "unable to initialize job template")
-		return &unstructured.Unstructured{}, err
+		return nil, err
 	}
-	unstructuredJob, err := templates.ProcessTemplate(context.TODO(), instance, jobTemplate)
+
+	job, err := processJobTemplate(jobTemplate, instance)
 	if err != nil {
-		log.Error(err, "unable to process", "template", jobTemplate, "with parameter", instance)
-		return &unstructured.Unstructured{}, err
+		log.Error(err, "unable to process job template")
+		return nil, err
 	}
-	return unstructuredJob, nil
+
+	return job, nil
+}
+
+func processJobTemplate(jobTemplate string, instance *mustgatherv1alpha1.MustGather) (*batchv1.Job, error) {
+	job := &batchv1.Job{}
+	err := yaml.Unmarshal([]byte(jobTemplate), job)
+	if err != nil {
+		return nil, fmt.Errorf("failed to umarshal job template: %w", err)
+	}
+	job.ObjectMeta.Name = instance.ObjectMeta.Name
+	job.ObjectMeta.Namespace = instance.ObjectMeta.Namespace
+
+	var gatherCommandBinary string
+	if audit := instance.Spec.Audit; audit {
+		gatherCommandBinary = gatherCommandBinaryAudit
+	} else {
+		gatherCommandBinary = gatherCommandBinaryNoAudit
+	}
+
+	job.Spec.Template.Spec.Containers[0].Command = append(
```

Comment:
For context to the earlier comment, I'm mostly concerned about the brittleness of these array indices if someone comes in and changes the template in some way. My concerns could also be addressed with unit tests that would fail if someone mucked with the jobTemplate's containers

Reply:
+1 this comment fully convinces me we should address this now. There's no guarantee of order in the unmarshalling.